pynicotine/shares.py

```diff
@@ -477,14 +477,15 @@ def is_hidden(folder, filename=None, entry=None):
         """Stop sharing any dot/hidden folders/files."""
 
         # If the last folder in the path starts with a dot, or is a Synology extended
-        # attribute folder, we exclude it
+        # attribute, snapshot or recycle bin folder, we exclude it
         if filename is None:
             last_folder = os.path.basename(folder)
 
-            if last_folder.startswith(".") or last_folder == "@eaDir":
+            HIDDEN_FOLDER_NAMES = {"@eaDir", "#recycle", "#snapshot"}
```

Comment:
Please move this to the beginning of the class, below the initial comment (line 232).

Reply:
I've moved it below this:
```
class Scanner:
    """Separate process responsible for building shares.

    It handles scanning of folders and files, as well as building
    databases and writing them to disk.
    """
```